app/src/main/java/com/moyerun/moyeorun_android/common/BadgeRoundImageView.kt

```diff
@@ -0,0 +1,94 @@
+package com.moyerun.moyeorun_android.common
+import android.content.Context
+import android.content.res.TypedArray
+import android.util.AttributeSet
+import android.view.LayoutInflater
+import androidx.constraintlayout.widget.ConstraintLayout
+import com.moyerun.moyeorun_android.R
+import com.moyerun.moyeorun_android.databinding.ViewBadgeRoundImageBinding
+
+class BadgeRoundImageView : ConstraintLayout {
+    constructor(context: Context): super(context) { initView() }
+    constructor(context: Context, attrs: AttributeSet): super(context, attrs) {
+        initView()
+        getAttrs(attrs)
+    }
+    constructor(context: Context, attrs: AttributeSet, defStyle: Int): super(context, attrs) {
+        initView()
+        getAttrs(attrs, defStyle)
```

Comment:
getAttrs() 메소드 내에서 setTypedArray() 로직이 포함되어 코드 문맥상 어색한 느낌이 듭니다!

constructor 내에서도 1. 뷰를 초기화 한다. 2. Attribute를 얻는다. 라고 이해되는데 내부 로직은 세팅까지 모두 끝나는 로직이니 Attrs를 가져오기만 했는데 세팅까지 되어버리네? 라고 이해되어 코드만 보고 흐름을 예상하기가 어색한 것 같습니다.

차라리, set~() 메소드 내에서 get~()을 호출하고 + 세팅까지 마치는 로직을 포함시키고 constructor 내에서는 initView() 와 set~()을 하는 것이 어떨까요~?! 이렇게 하면 뷰를 초기화 시킨 후 attrs 세팅을 하는구나 라고 이해할 수 있을 것 같습니다!
```suggestion
        initView()
        set~()
```

Reply:
말씀하신대로 get 부분에서 혼동의 여지가 있을 수 있을것 같습니다.
Attribute를 기본값까지 설정하여 사용할 수 있다는 의미로 setAttrs(attrs, defStyle)로 변경한 뒤 
내부에서 받아온 attrs를 통해 setTypedArray()을 사용하는 방식으로 수정해놓도록 하겠습니다!